src/TheThingsNetwork.cpp

```diff
@@ -401,13 +401,12 @@ void TheThingsNetwork::autoBaud()
 
 void TheThingsNetwork::reset(bool adr)
 {
-  if (!baudDetermined)
-  {
-    autoBaud();
-  }
-
+  autoBaud();
   size_t length = readResponse(SYS_TABLE, SYS_RESET, buffer, sizeof(buffer));
 
+  autoBaud();
+  length = readResponse(SYS_TABLE, SYS_TABLE, SYS_GET_VER, buffer, sizeof(buffer));  
```

Comment:
I did not notice this call in the autobaud ... in my opinion it's a bit out of place to call the "sys get ver" in the autobaud, I would be more to delete it in that function.